### PR TITLE
Added sensor state trait.  

### PR DESCRIPTION
**Pull request by wir3z**
Added sensor state trait to allow querying Co, CO2, smoke, rain, filter life, etc sensors for descriptive state as well as numerical value if applicable.

Quantized the set temperature to 0.1 degrees to address Google's large multiple decimal place floating point math.

### CODE REVIEW

File: google-home-community.groovy [hunk 1]
```diff
@@ -2628,6 +2682,9 @@ private executeCommand_ThermostatTemperatureSetpoint(deviceInfo, command) {
         setpoint = celsiusToFahrenheitRounded(setpoint)
     }
 
+    // round to 0.1 degree resolution
+    setpoint = (setpoint.toFloat()*10).toInteger()/10
```
Comment (mbudnek) [inline google-home-community.groovy:2686]: This seems unrelated.  Also `setpoint` should already be rounded to one decimal place.

File: README.md [hunk 2]
```diff
@@ -134,11 +130,14 @@ The Arm/Disarm trait is user to control security systems.  It can be controlled
 	- Google Home Level Names for Away: Comma separated list of synonyms that Google can send to set this level.  Maps to the `Away` attribute by default.
 	- Hubitat Command for Away: The device command used to set this level.  Maps to the `armAway` attribute by default.
 	- Hubitat Value for Away: The device value returned when the device is set to this level.  Maps to the `armed away` attribute by default.
+- Use device pincodes: Set to retrieve the pincodes from the device.  Set to false to use the app pincodes.  Defaults to 'false'.
+  NOTE: Global pincodes are always useable.
+	- Device pin code attribute: The device attribute used to query the device pincodes JSON map.  Maps to the `lockCodes` attribute by default.
+	- Device pin code value: The key value in the JSON map pointing to the user pincode.  Maps to the `code` value by default.
```
Comment (mbudnek) [inline README.md:136]: I moved this to the "Device Type Settings" section.  Please merge master.

File: README.md [hunk 3]
```diff
@@ -254,8 +253,10 @@ The Lock/Unlock trait is used for anything that can lock and unlock, such as doo
 - Locked Value: The value that the Locked/Unlocked attribute will report when the device is locked.  Defaults to "locked".
 - Lock Command: A device command used to lock the device.  Should not require any parameters.  Maps to `lock` by default.
 - Lock Command: A device command used to unlock the device.  Should not require any parameters.  Maps to `unlock` by default.
-- Return user index: If selected, during a successful pincode challenge, the user index is returned with the Arm/Disarm commands. Defaults to 'false'.
-	  NOTE: Device must support parameters in the lock / unlock commands.
+- Use device pincodes: Set to retrieve the pincodes from the device.  Set to false to use the app pincodes.  Defaults to 'false'.
+  NOTE: Global pincodes are always useable.
+	- Device pin code attribute: The device attribute used to query the device pincodes JSON map.  Maps to the `lockCodes` attribute by default.
+	- Device pin code value: The key value in the JSON map pointing to the user pincode.  Maps to the `code` value by default.
```
Comment (mbudnek) [inline README.md:259]: Same here

**Review by mbudnek — COMMENTED** [review]
Looks great.  Just a few small comments

File: README.md (hunk 2 above)
Comment (wir3z) [reply README.md]: Fixed.

File: README.md (hunk 3 above)
Comment (wir3z) [reply README.md]: Fixed.

File: google-home-community.groovy (hunk 1 above)
Comment (wir3z) [reply google-home-community.groovy:2686]: Not sure if its because I'm using C instead of F?  The set points coming back from GHC to Hubitat in this case have 10 decimal place precision.

I can take this out if you like (or different PR)?

Comment (mbudnek) [reply google-home-community.groovy:2686]: I see.  Are you setting the thermostat in F in Google Home, but using C in Hubitat?  I see Google does the conversion out to many decimal places.  I'll make a separate PR to clean that up.

**Review by mbudnek — APPROVED** [review]

**Comment by mbudnek** [comment]
Merged.  Thanks again for the contribution!
